usr/sbin/rear

```diff
@@ -729,6 +762,17 @@ if test "$DEBUGSCRIPTS_ARGUMENT" ; then
     Debug "The debugscripts flags are '$DEBUGSCRIPTS_ARGUMENT'"
 fi
 
+# Show TRUSTWORTHY_OWNERS and TRUSTWORTHY_PATHS before files get sourced:
+Debug "Sourced files must be owned by one of the TRUSTWORTHY_OWNERS ${TRUSTWORTHY_OWNERS[*]}"
+Debug "Sourced files must be below one of the TRUSTWORTHY_PATHS ${TRUSTWORTHY_PATHS[*]}"
+
+# Include functions after RUNTIME_LOGFILE is set and readonly
+# so that functions can use a fixed RUNTIME_LOGFILE value
+# and after the initial startup messages and TRUSTWORTHY_OWNERS and TRUSTWORTHY_PATHS are shown:
+for script in $SHARE_DIR/lib/[a-z]*.sh ; do
+    source $script || BugError "Failed to source $script"
```

Comment:
So this `source()`  call is already using the one from `_input_output_functions.sh`? I find this highly confusing that we first define `source()` to throw an error, and *then* we override that with a new definition of `source()` that doesn't throw an error but performs trustworthy checks.

And, we also have `Source()` (with capital S) which again does something different.

Is there a way to avoid redefining `source()` twice and instead using our existing `Source()` function (after adding the trustworthy checks to it) - I think that would reduce the mental complexity for us and other potential contributors.

Reply:
The point is not what we do to Source() our own scripts.
The point is what arbitrary contributions 'source'
for third-party software support.
So the point is to gain control over what calls 'source'.